### PR TITLE
feat(validator): Adding error, warn, info categories support.

### LINKED ISSUE #168: feat(validator): validation categories: error, warning, info
Not all validation result should be an error. Think of weak passwords. Or some warning that the value is near a threshold. 

Implement a `category` prop to setting up validators that, when invalid is triggered, will set the correct attribute. 

Right now we set `data-validation-haserror`. The goal is to have other two
- `data-validation-warning`
- `data-validation-info`

`showValidationHints` will now render orange for warning and green for info

Like traffic lights. 😁

🔴🟠🟢

### PR DESCRIPTION
**Pull request by Lalit3716**
## feat(validator): Adding various categories support for validators like error, warn, info etc...

Fixes #168  

Description of changes: 
- Added ValidationRule type for better type safety
- Accepting either strings or objects with categories as valid parameters for validators

Tag a reviewer: @ayoayco 

Tasks:
- [x] I have ran the build command to make sure apps are working: `npm run build`
- [x] I have ran the tests to make sure nothing is broken: `npm run test`
- [x] I have ran the linter to make sure code is clean: `npm run lint`
- [x] I have reviewed my own code to remove changes that are not needed

### CODE REVIEW

File: packages/validator/core/validator-functions.ts [hunk 1]
```diff
@@ -7,42 +7,74 @@ import { Validators } from './validator-names';
...
 			// TODO: implement dynamic import of function depending on validators
 			const split = attribute.split(':');
 			const validator = split[0];
 			const limitStr = split[1];
 			const limit = parseInt(limitStr || '0', 10);
 
 			if (validator === Validators.min()) {
-				return validateMin(value, limit);
+				const val = validateMin(value, limit);
+				if (val) {
```
Comment (ayoayco) [inline packages/validator/core/validator-functions.ts:31]: Generally, you want to minimize the if-within-an-if to reduce the function complexity.

I would add an optional parameter in the validation functions, like so: `validateMin(value, limit, category)` then add the `category` prop in the object they return :)

Comment (Lalit3716) [reply packages/validator/core/validator-functions.ts:31]: Right sure! I am just trying out different things and seeing what will work best!

**Comment by Lalit3716** [comment]
Hi @ayoayco, finally I fixed tests/lint issues with this PR 😅, it's ready to be merged now! Although I haven't added test cases for various categories of validators, I think I will open a separate PR for that if that's ok with you.

**Review by ayoayco — APPROVED** [review]
Alright! Thanks for this @Lalit3716 🚀

**Pull request merged**